api/src/mailbox.rs

```diff
@@ -1047,6 +1051,70 @@ pub struct AuthorizeAndStashResp {
 }
 impl Response for AuthorizeAndStashResp {}
 
+pub fn mbox_read_fifo(
+    mbox: mbox::RegisterBlock<impl MmioMut>,
+    mut buf: &mut [u8],
+) -> core::result::Result<(), CaliptraApiError> {
+    use zerocopy::Unalign;
+
+    fn dequeue_words(mbox: &mbox::RegisterBlock<impl MmioMut>, buf: &mut [Unalign<u32>]) {
+        for word in buf.iter_mut() {
+            *word = Unalign::new(mbox.dataout().read());
+        }
+    }
+
+    let dlen_bytes = mbox.dlen().read() as usize;
+
+    if dlen_bytes < buf.len() {
+        buf = &mut buf[..dlen_bytes];
+    }
+
+    let len_words = buf.len() / size_of::<u32>();
+    let (mut buf_words, suffix) = LayoutVerified::new_slice_unaligned_from_prefix(buf, len_words)
+        .ok_or(CaliptraApiError::ReadBuffTooSmall)?;
+
+    dequeue_words(&mbox, &mut buf_words);
+    if !suffix.is_empty() {
+        let last_word = mbox.dataout().read();
+        let suffix_len = suffix.len();
+        suffix
+            .as_bytes_mut()
+            .copy_from_slice(&last_word.as_bytes()[..suffix_len]);
+    }
+
+    Ok(())
+}
+
+pub fn mbox_write_fifo(
+    mbox: &mbox::RegisterBlock<impl MmioMut>,
+    buf: &[u8],
+) -> core::result::Result<(), CaliptraApiError> {
+    const MAILBOX_SIZE: u32 = 128 * 1024;
```

Comment:
This constant appears in about a dozen places. Can we pull it from somewhere rather than creating another constant?

Reply:
@swenson , In my follow-up to this PR, I tried to pull in the constant from the caliptra-drivers crate. For some convoluted reason I can't pretend to understand, the hash of caliptra-rom changed.